Purchases/Public/RCPurchases.m

```diff
@@ -910,12 +910,18 @@ - (void)handleOfferingsUpdateError:(NSError *)error completion:(RCReceiveOfferin
     CALL_IF_SET_ON_MAIN_THREAD(completion, nil, error);
 }
 
-- (void)receiptData:(void (^ _Nonnull)(NSData * _Nonnull data))completion
-{
+- (void)receiptData:(void (^ _Nonnull)(NSData * _Nonnull data))completion {
+    [self receiptDataWithForceRefresh:false completion:completion];
+}
+
+- (void)receiptDataWithForceRefresh:(BOOL)forceRefresh completion:(void (^ _Nonnull)(NSData * _Nonnull data))completion  {
     NSData *receiptData = [self.receiptFetcher receiptData];
     if (receiptData == nil) {
```

Comment:
I wonder if we should a zero-bytes check here too. I've seen that before, where there is a file, but it's empty.

Reply:
Possible there is a check in the fetcher.

Reply:
We have the check in a couple of places in the RCPurchases.m:
https://github.com/RevenueCat/purchases-ios/blob/cesardelavega/ch68/force-refresh-receipt/Purchases/Public/RCPurchases.m#L629
https://github.com/RevenueCat/purchases-ios/blob/cesardelavega/ch68/force-refresh-receipt/Purchases/Public/RCPurchases.m#L1084

I will add it a check when we fetch so we also log a message in that case and also here so that we also refresh if the receipt has 0 bytes.